src/BenchmarkDotNet.Diagnostics.Windows/Tracing/NativeMemoryLogParser.cs

```diff
@@ -91,7 +117,27 @@ public IEnumerable<Metric> Parse()
                         nativeLeakSize += data.AllocSize;
                         totalAllocation += data.AllocSize;
                     }
+
+                    bool IsCallStackIn(StackSourceCallStackIndex index)
+                    {
+                        var frame = stackSource.GetFrameIndex(index);
+                        var name = stackSource.GetFrameName(frame, false);
+                        var stack = name.Split(new []{'!'}, StringSplitOptions.RemoveEmptyEntries);
+                        if (stack.Length == 2 &&
+                            benchmarkedFunctionStacks.Any(s=>
+                            string.Equals(stack[0], s.module, StringComparison.OrdinalIgnoreCase) &&
+                            s.function.IsMatch(stack[1])))
+                        {
+                            return true;
+                        }
+
+                        var callerIndex = stackSource.GetCallerIndex(index);
+
+                        return callerIndex != StackSourceCallStackIndex.Invalid
+                               && IsCallStackIn(callerIndex);
+                    }
```

Comment:
this method is quite expensive. For every stack in the stack trace we are:

* calling the method in a recursive way (we should be able to avoid `StackOverflowException` because some process was able to get such a deep call stack)
* allocating an array of strings (`string.Split`)
* comparing the name using 3 regular expressions

> I wanted to fix it by running a benchmark twice but I got a different JIT event.

This was my first idea, it would be great if we could just ignore the results from the first iteration ;)

But if this is impossible, perhaps we could just search for `.WorkloadAction` in the call stacks? Sth like this:

```cs
while (index != StackSourceCallStackIndex.Invalid)
{
    var frame = stackSource.GetFrameIndex(index);
    var name = stackSource.GetFrameName(frame, false);
    
    if (name.IndexOf(".WorkloadAction", StringComparison.Ordinal) > 0)
    {
        return true;
    }
    
    index = stackSource.GetCallerIndex(index);
}

return false;
```

Reply:
I need more time to investigate it. Each idea takes a long time to test it. I have 12 benchmark in my test repo https://github.com/WojciechNagorski/NativeMemorySample and I have to run it for .Net Framework and .Net Core.

Reply:
@adamsitnik I've used your iterative algorithm with minor modifications.